pkg/storage/snapshot/source.go

```diff
@@ -179,6 +184,69 @@ func (s *vmSnapshotSource) getVMRevision() (*snapshotv1.VirtualMachine, error) {
 	return vmRevision, nil
 }
 
+func (s *vmSnapshotSource) getControllerRevision(namespace, name string) (*appsv1.ControllerRevision, error) {
+	crKey := cacheKeyFunc(namespace, name)
+	obj, exists, err := s.controller.CRInformer.GetStore().GetByKey(crKey)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, fmt.Errorf("Unable to fetch ControllerRevision %s", crKey)
+	}
+	return obj.(*appsv1.ControllerRevision), nil
```

Comment:
Check for successful cast?

Reply:
I can to be complete but I'm not sure that it's possible for this informer to return anything other than this type (for now at least).

Reply:
I think it's still worth checking for the cast to be successful. Ideally there would be a policy covering the whole project whether unchecked cast should be allowed or not.